cairo-hints/scarb-hints-run/main.rs

```diff
@@ -206,8 +199,74 @@ fn run() -> Result<String, Box<dyn std::error::Error>> {
         args.proof_mode,
     );
 
+    process_result(result, postprocess_config).await
+}
+
+async fn get_func_args(
+    args: &Args,
+    package: &scarb_metadata::PackageMetadata,
+    preprocess_config: Option<ServerConfig>,
+) -> Result<FuncArgs> {
+    let inputs_schema = get_inputs_schema(package)?;
+    let schema = parse_input_schema(&inputs_schema)
+        .map_err(|e| anyhow::anyhow!("Failed to parse input schema: {}", e))?;
+
+    if let Some(preprocess) = preprocess_config {
+        let body: Value =
+            serde_json::from_str(&args.args_json.as_ref().context("Expect --args_json")?)?;
+
+        let url = if preprocess.server_url.ends_with("/preprocess") {
+            preprocess.server_url.to_string()
+        } else if preprocess.server_url.ends_with('/') {
+            format!("{}preprocess", preprocess.server_url)
+        } else {
+            format!("{}/preprocess", preprocess.server_url)
+        };
+
+        let preprocess_result = call_server::<PreprocessResponse>(&url, body).await?.args;
+        process_json_args(&preprocess_result, &schema).map_err(|e| anyhow::anyhow!(e))
+    } else if let Some(json_args) = &args.args_json {
+        process_json_args(json_args, &schema).map_err(|e| anyhow::anyhow!(e))
+    } else if let Some(args) = &args.args {
+        Ok(args.clone())
+    } else {
+        Ok(FuncArgs::default())
+    }
+}
+
+fn get_inputs_schema(package: &scarb_metadata::PackageMetadata) -> Result<PathBuf> {
+    absolute_path(package, None, "inputs_schema", Some(PathBuf::from("InputsSchema.txt")))
+        .context("Inputs schema path must be provided either in the Scarb.toml file in the [tool.hints] section or default to InputsSchema.txt in the project root.")
+}
+
+async fn process_result(
+    result: Result<Option<String>, Error>,
+    postprocess_config: Option<ServerConfig>,
+) -> Result<String> {
     match result {
-        Ok(return_values) => Ok(return_values.unwrap_or_else(|| "Null".to_string())),
+        Ok(return_values) => {
+            let cairo_output = return_values.unwrap_or_else(|| "Null".to_string());
+            if let Some(postprocess) = postprocess_config {
```

Comment:
In `orion-runner` I added the possibility of specifying this URLs throw [env variables](https://github.com/gizatechxyz/orion_runner/blob/a8f868d97e312acd62afaa4bcb161734c188ddbc/crates/platform-utils/src/utils.rs#L198) rather than adding them to the server. By default they will be in localhost but changed if need so we dont have to make changes to the `servers.json` WDYT?

Reply:
Good point! I changed it! However I used port `3000` if you don't mind as it's the one generated by the template when we do `scarb hints-new`.